tables/Tables/ArrayColumn.h

```diff
@@ -501,6 +504,45 @@ class ArrayColumn : public TableColumn
     mutable Bool reaskAccessColumnSlice_p;
 };
 
```

Comment:
I'll add some more comments and also have it do some validation on 
creation; that will be better than letting it fail the first time it is 
used in getColumnCells.

(It's possible I'm answering the wrong question here.)  I don't think 
it's good practice to create complex objects visible in the API in an 
/ad hoc /fashion using simple data structures (e.g., the 
Vector<Vector<Slice>> that was in the previous logic, although that 
worked slightly differently). Important data structure should be 
objects.  Perhaps instead of two vectors I should have used a single 
vector of a simple object which held one destination and one source 
slicer?  CASA proper is littered with parallel arrays which probably 
came from Fortran programming idioms; modern usage would be a single 
array of objects.

On 06/25/2015 12:25 AM, Ger van Diepen wrote:
>
> In tables/Tables/ArrayColumn.h 
> <https://github.com/casacore/casacore/pull/121#discussion_r33225529>:
>
> > @@ -501,6 +504,45 @@ class ArrayColumn : public TableColumn
> >      mutable Bool reaskAccessColumnSlice_p;
> >  };
> >
>
> I'd like some comments in class ColumnSlicer. Furthermore, a test 
> program seems to be lacking.
> Furthermore, I wonder why ColumnSlicer has a Vector. It raises the 
> question why it is not , but also: why not Vector? Do you think that 
> constructing it is too expensive?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/casacore/casacore/pull/121/files#r33225529>.Web 
> Bug from 
> https://github.com/notifications/beacon/AGPGtKLtwE74xcGIV5hUR00adPRzKoNYks5oW5ZxgaJpZM4FDwg4.gif
>

-- 
Jim Jacobs

phone: 835-7235
office: 301